indexing-service/src/main/java/io/druid/indexing/overlord/TaskQueue.java

```diff
@@ -499,14 +506,25 @@ private void syncFromStorage()
     try {
       if (active) {
         final List<Task> newTasks = taskStorage.getActiveTasks();
+        List<Task> addedTasks = Lists.newArrayList(Sets.difference(Sets.newHashSet(newTasks), Sets.newHashSet(tasks)));
+        List<Task> removedTasks = Lists.newArrayList(Sets.difference(Sets.newHashSet(tasks), Sets.newHashSet(newTasks)));
+
+        // Clean up removed Tasks
+        for(Task task : removedTasks){
+          shutdown(task.getId());
```

Comment:
The idea being that we are trying to keep ```taskLockbox``` and ```tasks``` in sync, but the shutting down of things involves the runner too, and that's not something we can really try to keep in sync with the other two because it is asynchronous.